lib/spack/spack/build_environment.py

```diff
@@ -522,9 +523,12 @@ def _set_variables_for_single_module(pkg, module):
     m.configure = Executable('./configure')
 
     m.meson = Executable('meson')
+    # m.cmake = Executable('cmake')
     m.cmake = Executable('cmake')
     m.ctest = MakeExecutable('ctest', jobs)
 
+    if os.name == 'nt':
```

Comment:
I see some checks like this and some checks using `sys.platform`, can we use the latter universally? 

* `win32` is more understandable than `nt` IMO
* Either way, I think we should do the same thing everywhere

Reply:
This has been an ongoing pain point. The issue is essentially that while, `sys.platform` returns 'win32' on CMD prompts and PWSH terminals, using a MSYS shell, cygwin, WSL, or some other 'unix environment in Windows' project, it returns other values. I believe the same is true of `os.name` with slightly less variation than `sys.platform`. Betsy and I at one point put together an effort to refactor all of these changes to use Spack's internal platform checking, but this blew up on testing, where the platform change to `Testing` broke all the checks. That work was then reverted, but still exists somewhere. We can work a solution from that angle, or just quick and easy grep for all the uses of `os.name` and replace them.

Reply:
Could this be wrapped in a function that can check for the different possible values of `sys.platform` (and then that one function can be called everywhere)?

Reply:
Absolutely, if that's the direction we want to take with it. We did initially want to use Spacks internal checking system to provide a little more robustness/consistency for the check rather than checking against a pre-baked list, but if you think that's not necessary, then, absolutely, no problem.

Reply:
Addressed in review commit.